holidays/countries/jersey.py

```diff
@@ -0,0 +1,246 @@
+        # Prior to 2004, in-lieu are only given for Sundays.
+        # https://www.jerseylaw.je/laws/enacted/Pages/RO-123-2004.aspx
+        kwargs.setdefault(
+            "rule", SUN_TO_NEXT_MON if dt < date(2004, OCT, 12) else self._observed_rule
+        )
+        return super()._add_observed(dt, **kwargs)
+
+    def _populate_public_holidays(self) -> None:
+        # Earliest available piece of law available is from 1952.
+        if self._year <= 1951:
+            return None
+
+        # Good Friday
+        self._add_good_friday("Good Friday")
+
+        # May Day
+        # This only starts in 1980 (instead of 1978) for Jersey.
+        # The date is not moved in 2020 (unlike in the UK) as there's already VE Day Celebrations.
+
+        # May Day bank holiday (first Monday in May)
+        if self._year >= 1980:
+            name = "May Day"
+            if self._year == 1995:
+                self._add_holiday_may_8(name)
+            else:
+                self._add_holiday_1st_mon_of_may(name)
+
+        # Spring bank holiday
+        # Current Pattern started in 1970 for Jersey.
+
+        # Spring bank holiday (last Monday in May)
+        if self._year >= 1970:
+            spring_bank_dates = {
+                2002: (JUN, 4),
+                2012: (JUN, 4),
+                2022: (JUN, 2),
+            }
+            name = "Spring Bank Holiday"
+            if self._year in spring_bank_dates:
+                self._add_holiday(name, spring_bank_dates[self._year])
+            else:
+                self._add_holiday_last_mon_of_may(name)
+
+        # Whit Monday.
+        # Was in-use prior to Spring bank holiday adoption.
+
+        if self._year <= 1969:
+            # Whit Monday.
+            self._add_whit_monday("Whit Monday")
+
+        # Easter Monday
+        self._add_easter_monday("Easter Monday")
+
+        # Late Summer Bank Holiday
+        # Current Pattern started in 1970. Was previously first Monday of September for Jersey.
+
+        # Late Summer bank holiday (last Monday in August)
+        summer_bank_holiday = "Late Summer Bank Holiday"
+        if self._year >= 1970:
+            self._add_holiday_last_mon_of_aug(summer_bank_holiday)
+        else:
+            self._add_holiday_1st_mon_of_sep(summer_bank_holiday)
+
+        # Jersey exclusive holidays
+
+        # Liberation Day.
+        # Started in 1952. This has no in-lieus.
+        # Counts as Public Holiday when fall on the weekdays, also on Saturday from 2010 onwards.
+        # Specially held in 2010 on Sunday for the 65th Anniversary.
+
+        if self._year >= 1952:
+            # Liberation Day
+            liberation_day = self._add_holiday_may_9("Liberation Day")
+            if (self._is_sunday(liberation_day) and self._year != 2010) or (
+                self._is_saturday(liberation_day) and self._year <= 2010
+            ):
+                self.pop(liberation_day)
+
+    def _populate_subdiv_holidays(self):
```

Comment:
As we are not inherited from UK, we can just add these holidays in `_populate_public_holidays()`

Reply:
Thanks for the catch, I forgot to change them around after I realized reusing `GB`'s  `_populate_public_holidays` no longer perfectly matches `JE` law.

`GB` could probably get a revisit since the "Whit Monday" holidays (precursor to the current Spring bank holiday) from 1871 law wasn't implemented yet.